File: src/portfolio_manager/models/stock_price.py
"""Stock price cache model."""

from dataclasses import dataclass
from datetime import date, datetime
from decimal import Decimal
from uuid import UUID


@dataclass
class StockPrice:
    """Cached stock price for a given date."""

    id: UUID
    ticker: str
    price: Decimal
    currency: str
    name: str
    exchange: str | None
    price_date: date
    created_at: datetime
    updated_at: datetime

File: internal/models/stock_price.go
package models

import (
	"database/sql"

	"github.com/kadragon/portfolio-manager/internal/datex"
	"github.com/kadragon/portfolio-manager/internal/ktime"
	"github.com/kadragon/portfolio-manager/internal/numeric"
	"github.com/kadragon/portfolio-manager/internal/uuidx"
)

// StockPrice is a cached price entry for a ticker on a given date.
type StockPrice struct {
	ID        uuidx.UUID
	Ticker    string
	Price     numeric.Decimal
	Currency  string
	Name      string
	Exchange  sql.NullString
	PriceDate datex.Date
	CreatedAt ktime.Time
	UpdatedAt ktime.Time
}
